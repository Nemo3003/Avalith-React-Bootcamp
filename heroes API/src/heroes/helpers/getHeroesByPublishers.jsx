import React from 'react'
import {heroes } from '../data/heroesData'

export const getHeroesByPublishers = (publisher) => {

    const validaPublisher = ['DC Comics', ['Marvel Comics']]

    if (!validaPublisher.includes(publisher)) {
        throw new Error(`${publisher} is not a valid publisher`)
    }
  return (
    <div>
        <ul>
            {
                heroes.map(
                    hero => {
                        if (hero.publisher === publisher) {
                            return <li key={hero.id}>{hero.name}</li>
                        }
                    }
                )
            }
        </ul>
    </div>
  )

        }