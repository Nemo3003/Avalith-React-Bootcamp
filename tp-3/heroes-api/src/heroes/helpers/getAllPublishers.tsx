import React from "react";

export const getAllPublishers = (publicAll:any) => {
  const publishers:any[]  = [];
  for (let i = 0; i < publicAll.length; i++) {
    if (
      !publishers.includes(publicAll[i].biography.publisher) &&
      publicAll[i].biography.publisher !== null && publicAll[i].biography.publisher !== ""
    ){
      publishers.push(publicAll[i].biography.publisher);}}

  return publishers;};
