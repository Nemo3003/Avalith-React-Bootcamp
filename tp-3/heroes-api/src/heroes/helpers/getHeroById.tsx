export const getHeroById = (heroId:any, arr:any) => {
    return (
      arr.find((hero:any) => 
      hero.id === parseInt(heroId)));
  };
  