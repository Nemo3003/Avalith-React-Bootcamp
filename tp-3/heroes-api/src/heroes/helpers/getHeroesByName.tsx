export const getHeroesByName = (name = "", arr:any[]) => {
    name = name.toLowerCase().trim();
  
    if (name.length === 0) return [];
  
    return arr.filter((hero) => hero.name.toLowerCase().includes(name));
  };