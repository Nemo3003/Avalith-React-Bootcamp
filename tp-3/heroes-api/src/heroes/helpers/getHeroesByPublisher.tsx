export const getHeroesByPublishers = (publisher:String, arr:any[]) => {
    if (publisher === "ALL" || publisher === "") {
      return arr;
    } else {
      const finalArray = arr.filter(
        (hero) => hero.biography.publisher === publisher
      );
  
      if (finalArray) {
        return finalArray;
      } else {
        throw new Error(`${publisher} is not a valid publisher`);
      }
    }
  };
  