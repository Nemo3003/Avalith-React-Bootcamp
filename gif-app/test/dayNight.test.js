import {DayNight} from '../../src/helpers/dayNight';

describe('changing colors on the day and at night', function () {
    test('changing colors on the day and at night', function () {
        const day = jest.fn();
        const night = jest.fn();
        
    })
})