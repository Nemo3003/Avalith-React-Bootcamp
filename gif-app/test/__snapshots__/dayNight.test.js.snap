// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navigation"
>
  <a
    className="site-title"
    href="https://github.com/Nemo3003"
  >
    Ulises' Github
  </a>
  <ul>
    <li>
      <a
        href="#"
      >
        Home
      </a>
    </li>
    <li>
       
      <a
        href="#"
      >
         Practice
      </a>
    </li>
    <li>
       
      <a
        href="#"
      >
         First Project
      </a>
    </li>
    <li>
      <a
        href="#"
      >
         Second Project
      </a>
    </li>
    <li>
      <a
        href="#"
      >
         Third Project
      </a>
    </li>
  </ul>
</nav>
`;
